packages/react-examples/src/react/FocusTrapZone/e2e/FocusTrapZone.e2e.ts

```diff
@@ -1,24 +1,413 @@
+import type { FTZTestWindow } from './shared';
+
 const ftzStoriesTitle = 'Components/FocusTrapZone/e2e';
 
+/**
+ * Calls `window.setProps()` -- this must be defined by the story being tested
+ */
+function setProps(props: IFocusTrapZoneProps) {
+  cy.window().then(win => (win as FTZTestWindow).setProps!(props));
+}
+
 describe('FocusTrapZone', () => {
   before(() => {
     cy.visitStorybook({ qs: { e2e: '1' } });
   });
 
+  // These are basic tests of different props, but due to the reliance on focus behavior they're
+  // best done in the browser.
+  describe('Focus behavior based on default and explicit prop values', () => {
+    beforeEach(() => {
+      cy.loadStory(ftzStoriesTitle, 'PropValues');
+    });
+
+    it('Restores focus to FTZ when clicking outside FTZ', () => {
+      setProps({});
+
+      cy.get('#mid').focus();
+      cy.focused().should('have.id', 'mid');
+
+      // try to focus on button outside FTZ
+      cy.get('#before').focus();
+      // it focuses first button inside FTZ instead
+      cy.focused().should('have.id', 'first');
+    });
+
+    it('Does not restore focus to FTZ when clicking outside FTZ with isClickableOutsideFocusTrap', () => {
+      setProps({ isClickableOutsideFocusTrap: true });
+
+      // verify we can focus on button outside FTZ
+      cy.get('#before').focus();
+      cy.focused().should('have.id', 'before');
+    });
+
+    it('Focuses first element when focus enters FTZ does not have focus and first bumper receives focus', () => {
+      setProps({ disableFirstFocus: true, isClickableOutsideFocusTrap: true });
+
+      cy.get('#before').focus();
+      cy.focused().should('have.id', 'before');
+
+      cy.realPress('Tab');
+      cy.focused().should('have.id', 'first');
+    });
+
+    it('Focuses last element when FTZ does not have focus and last bumper receives focus', () => {
+      setProps({ disableFirstFocus: true, isClickableOutsideFocusTrap: true });
+
+      cy.get('#after').focus();
+      cy.focused().should('have.id', 'after');
+
+      cy.realPress(['Shift', 'Tab']);
+      cy.focused().should('have.id', 'last');
+    });
+
+    it('Restores focus to FTZ when focusing outside FTZ', () => {
+      setProps({});
+
+      cy.get('#mid').focus();
+      cy.focused().should('have.id', 'mid');
+
+      cy.get('#after').focus();
+      cy.focused().should('have.id', 'first');
+    });
+
+    it('Does not restore focus to FTZ when forceFocusInsideTrap is false', () => {
+      setProps({ forceFocusInsideTrap: false });
+
+      cy.get('#after').focus();
+      cy.focused().should('have.id', 'after');
+    });
+
+    it('Focuses first on mount', () => {
+      setProps({});
+
+      cy.focused().should('have.id', 'first');
+    });
+
+    it('Does not focus first on mount with disableFirstFocus', () => {
+      setProps({ disableFirstFocus: true });
+
+      // For some reason this doesn't work:
+      // cy.focused().should('match', 'body');
```

Comment:
what's the actual focused item in that case ?

Reply:
The body. `document.activeElement` is `body` if you inspect it, but that assertion doesn't match for some reason.

Reply:
yeah but if you run the test in the cypres ui, it should tell you what the result of `cy.focused()` is, just curious, maybe we're matching the wrong thing.... or even what we're expecting to be focused isn't that

Reply:
IIRC it says focused() is body but it doesn't match that selector. Maybe I'm misunderstanding what `match` does but it seemed like it should match any CSS selector.

Reply:
perhaps this ? 

`cy.get('body').should('have.focus')`

Reply:
https://docs.cypress.io/api/commands/should#Focus

Reply:
nope (╯°□°）╯︵ ┻━┻

Reply:
Looking at the error messages for various attempted assertions and thinking about it some more, I *think* the issue is that it's checking focus with `:focus` rather than looking at `document.activeElement`. Apparently even if `body` is `document.activeElement` it doesn't get `:focus`. https://codepen.io/ecraig12345/pen/jOavBRw

Since `cy.focused()` returns a jquery object with the matches, `cy.focused().should('have.length', 0)` works.

Reply:
Oh a bit better: `cy.focused().should('not.exist')`